sync/sync.go

```diff
@@ -341,7 +345,15 @@ Register:
 	logrus.Infof("Adding replica %s in WO mode", replicaAddress)
 	_, err = t.client.CreateReplica(replicaAddress)
 	if err != nil {
-		return err
+		logrus.Errorf("Failed to create replica, error: %s", err.Error())
+		// cases for above failure:
+		// - controller is not reachable
+		// - replica is already added (remove replica in progress)
+		// - error while creating snapshot, to start fresh
+		// - replica might be in errored state
+		// - other replica might be in WO mode, and
+		// we can only have one WO replica at a time
+		goto Register
```

Comment:
what is the reason for this change ? this failure also leading to fatal is looking better.

Reply:
Reason is mentioned in the comments, crashing replica is not good in case, controller is not adding it because it can only add one "WO" replica at a time. If we are good with crash for this case then i can remove this.

Reply:
Then better put a sleep before retrying, otherwise it will max out the CPU

Reply:
ok

Reply:
ok